hw/data/common_project_cfg.hjson

```diff
@@ -35,7 +35,15 @@
   revision: '''{eval_cmd}
             COMMIT_L=`git rev-parse HEAD`; \
             COMMIT_S=`git rev-parse --short HEAD`; \
-            echo "Revision: [\`$COMMIT_S\`]({results_server_url_prefix}{repo_server}/tree/$COMMIT_L)";
+            REV_STR="Revision: [\`$COMMIT_S\`]({results_server_url_prefix}{repo_server}/tree/$COMMIT_L)"
+            if [ -d "hw/foundry" ]; then \
+              cd hw/foundry; \
+              COMMIT_FOUNDRY_S=`git rev-parse --short HEAD`; \
+              REV_STR_FOUNDRY="Foundry Revision: \`$COMMIT_FOUNDRY_S\`"
+              echo "$REV_STR, $REV_STR_FOUNDRY"; \
+            else \
+              echo "$REV_STR"; \
```

Comment:
Seems like this should not be in the else clause...

Reply:
So if the foundry repo is not present, only the revision string of the open repo should be printed out, hence I have put this in the else clause...

Reply:
If the foundry repo is present, then the open source repo rev will not be printed.

Reply:
hmm? but don't the foundry and open source repo represent different things? I thought you'd want to have both.

Reply:
so the idea is that if the foundry repo is present, both revisions are printed like this:
```
Revision: <something>, Foundry Revision: <something else>
```
otherwise, only the open source revision is printed
```
Foundry Revision: <something else>
```

Reply:
We should have both revisions in the report to facilitate debugging.

Reply:
yeah agreed

Reply:
Ohh $REV_STR on line 42 as well. Sorry I overlooked that part. I think I prefer to cleanly isolate the foundry part like this (sorry again, this is nitpicky):

```
            echo "$REV_STR"; \
            if [ -d "{proj_root}/hw/foundry" ]; then \
              COMMIT_FOUNDRY_S=`git -C {proj_root}/hw/foundry rev-parse --short HEAD`; \
              REV_STR_FOUNDRY="Foundry Revision: \`$COMMIT_FOUNDRY_S\`"; \
              echo "$REV_STR_FOUNDRY"; \
            fi  
```

Reply:
Hey Sri, I have tried this before, but it does not render the second string properly if we use two echo statements here.
If we want to avoid the else clause, how about we concatenate the strings (this would also be extendable to multiple if/fi blocks)?
```
            if [ -d "{proj_root}/hw/foundry" ]; then \
              COMMIT_FOUNDRY_S=`git -C {proj_root}/hw/foundry rev-parse --short HEAD`; \
              REV_STR="$REV_STR, Foundry Revision: \`$COMMIT_FOUNDRY_S\`"; \
            fi \
            echo "$REV_STR"; 
```

Reply:
I updated my earlier comment to add `\n\n` to force a line break. That IMO should work. `<br>` is another option. I think a single space should also be fine - they will render on the same line (which is what you were doing as well).

Reply:
Ohh I just realized echo does not process line breaks or `\n`s. You could replace `echo` with `printf`.

Reply:
Yeah using a newlines and `echo` leads to an inconsistent format (i.e., the second line is not using the same formatting as the first). However, it does work when using `printf` and `<br>`.

I've updated this and also renamed `Revision` to `GitHub Revision` to make the distinction between the two repos clearer.

Reply:
What you've done here is fine but, for completeness, if you want escaped characters with `echo`, you can use `echo -e`. For example:
```
$ echo -e 'foo\nbar\tbaz\x44x'
foo
bar     bazDx
```